### PR TITLE
Replaced the port selection input with an interactive menu using Spectre.Console.

### LINKED ISSUE #4: Interactive select for serial port
**Description:**
Change the current text-based selection of the serial port (see [this line](https://github.com/orbisat-oeiras/backend/blob/b3c35b4b7c501712b017cf81b15f3a9569fff1a2/Program.cs#L23)) to an interactive menu, which lists available ports and allows users to select one with the up/down arrow keys and return.

**Motivation:**
This feature provides a better user experience and reduces errors caused by typos.

**Implementation:**
 - Implement it from scratch - a simple menu shouldn't be too hard to implement, it's mostly a matter of capturing input key by key and re-rendering the menu
 - Use a UI library - easier, and better if more menus are added in the future

Also check [this thread](https://stackoverflow.com/questions/60767909/c-sharp-console-app-how-do-i-make-an-interactive-menu)

**Additional notes/comments**
As a bonus, adding colours to the menu would look pretty nice!

### PR DESCRIPTION
**Pull request by ordinho**
Not sure if this works as it's supposed to, since I only have one COM port showing up. Test if possible.
Will close #4 

EDIT: I messed up something with the commits and now the CancellationToken is also included in this request and in the interactive menu branch too

### CODE REVIEW

**Comment by levimcgomes** [comment]
I'll pull ASAP and see if I can fix the branches, hopefully some rebasing and squashing will do it. I might need to force push, in that case I'll let you know beforehand.

**Comment by levimcgomes** [comment]
Fixed. For future reference, I used an interactive rebase so only the right commit was left in each branch. Unfortunately, I'll have to force push for these changes to take effect.

File: Program.cs [hunk 1]
```diff
@@ -19,10 +20,12 @@ public static void Main(string[] args) {
...
 			// Get the name of the serial port where data is arriving
-			Console.WriteLine("Enter the name of the serial port where the APC220 module is connected.\nAvailable ports are:");
-			Console.Write(string.Concat(SerialPort.GetPortNames().Select(x => "\t" + x + "\n")) + "> ");
-			string serialPortName = Console.ReadLine()!;
-
+			var serialPortName = AnsiConsole.Prompt(
+				new SelectionPrompt<string>()
+					.Title("Please select the serial port where the [blue]APC220 module[/] is connected.")
+					.PageSize(10)
+					.AddChoices(SerialPort.GetPortNames().Select(x => x))
```
Comment (levimcgomes) [inline Program.cs:27]: `Select`ing with the identity does nothing - it should be removed.

File: Program.cs [hunk 2]
```diff
@@ -19,10 +20,12 @@ public static void Main(string[] args) {
...
+			);
```
Comment (levimcgomes) [inline Program.cs:28]: The menu is erased after selecting a port - either find a way to make it persist, with the selected option highlighted, or add a simple log afterwards.

**Review by levimcgomes — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 2 above)
Comment (ordinho) [reply Program.cs:29]: Done.

**Review by levimcgomes — APPROVED** [review]
Looking good! Also tested it with two devices plugged in, the UX is great!

**Pull request merged**